app/experimenter/reporting/api/v7/views.py

```diff
@@ -0,0 +1,140 @@
+import datetime
+from collections import defaultdict
+from statistics import median
+from urllib.parse import urljoin
+
+from django.conf import settings
+from rest_framework import views
+from rest_framework.response import Response
+
+from experimenter.reporting.constants import ReportLogConstants
+from experimenter.reporting.models import ReportLog
+
+
+class ReportDataView(views.APIView):
```

Comment:
You don't wanna use a ListAPIView and just get a bunch of this logic for free?

Reply:
current structure of the data is 
```